wallet_anchor.go

```diff
@@ -76,7 +76,7 @@ func (l *LndRpcWalletAnchor) FundPsbt(ctx context.Context, packet *psbt.Packet,
 
 	return tarogarden.FundedPsbt{
 		Pkt:               pkt,
-		ChangeOutputIndex: uint32(changeIndex),
+		ChangeOutputIndex: changeIndex,
```

Comment:
Double checking: The range for this is [-1,INT32_MAX], and by converting to a uint32 we were dropping the information about if we had a change output at all. Now we can short-circuit in PSBT adjustment and exclusion proof generation if we have no non-Taro change output.

Reply:
Exactly. The `-1` would've been converted to a very high number before which would've lead to an error anyway. With everything being `int32` we can remove some casts as well.